libs/core/langchain_core/tracers/event_stream.py

```diff
@@ -748,6 +794,7 @@ async def _astream_events_implementation_v2(
 
     # Assign the stream handler to the config
     config = ensure_config(config)
+    run_id = config.setdefault("run_id", uuid4())
```

Comment:
this ensures we have a known id for this run (instead of assigning it inside start callback